src/bin/pgcopydb/ld_store.c

```diff
@@ -0,0 +1,446 @@
+/*
+ * src/bin/pgcopydb/ld_store.c
+ *     Implementation of a CLI to copy a database between two Postgres instances
+ */
+
+#include <errno.h>
+#include <getopt.h>
+#include <inttypes.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include "parson.h"
+
+#include "copydb.h"
+#include "ld_stream.h"
+#include "lock_utils.h"
+#include "log.h"
+#include "parsing_utils.h"
+#include "pg_utils.h"
+#include "schema.h"
+#include "signals.h"
+#include "string_utils.h"
+
+
+/*
+ * ld_store_open_replaydb opens the current replaydb file if it already exists,
+ * or create a new replaydb SQLite file for processing the streaming data.
+ */
+bool
+ld_store_open_replaydb(StreamSpecs *specs)
+{
+	StreamContext *privateContext = &(specs->private);
+	DatabaseCatalog *replayDB = specs->replayDB;
+
+	if (!ld_store_current_filename(specs))
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
+	bool createReplayDB = IS_EMPTY_STRING_BUFFER(replayDB->dbfile);
+
+	/* if we don't have a replayDB filename yet, it's time to create it */
+	if (IS_EMPTY_STRING_BUFFER(replayDB->dbfile))
+	{
+		if (privateContext->timeline == 0)
+		{
+			log_error("BUG: ld_store_open_replaydb: timeline is zero");
+			return false;
+		}
+
+		sformat(replayDB->dbfile, MAXPGPATH, "%s/%08d-%08X-%08X.db",
+				specs->paths.dir,
+				privateContext->timeline,
+				LSN_FORMAT_ARGS(privateContext->startpos));
```

Comment:
The idea is to re-introduce file rotation later. It just seems to much to solve at once, so I'm not sure yet if I'll be able to introduce it within the PR or if it will have to wait. The file rotation will not split within a transaction this time, though, by design.

Reply:
Makes sense. Thanks @dimitri .